src/containers/AddTodoCheckBox.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react'
+import ReactDom from 'react-dom'
+import { connect } from 'react-redux'
+import { addTodo } from '../actions'
+import {Button, Input,Glyphicon} from 'react-bootstrap';
+
+//let AddTodo = ({todo_text, dispatch }) => {
+//    console.log(todo_text)
+//    const submitTodoButton =  <Button bsStyle="primary" type="submit"><Glyphicon glyph="plus" /></Button>;
+//
+//    let input = {value: ""};
+//
+//    let handleChange = (event) => {
+//        todo_text = event.target.value;
+//        console.log(todo_text)
+//    };
+//
+//    let todoSubmit = (event) => {
+//        event.preventDefault();
+//        if (!todo_text.trim()) {
+//            return;
+//        }
+//        dispatch(addTodo(todo_text));
+//        todo_text = "";
+//        console.log(todo_text);
+//    };
+//
+//    return (
+//    <div>
+//      <form onSubmit={todoSubmit}>
+//        <Input bsSize="large" type="text" placeholder="Todo item..." onChange={handleChange} defaultValue={todo_text} buttonAfter={submitTodoButton} />
+//      </form>
+//    </div>
+//    )
+//};
+
+class AddTodo extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+
+         isChecked:false,
+          input: {
+
+              val: "TODO 1"
+          }
+        };
+    }
+
+    todoSubmit = (event) => {
+        event.preventDefault();
+        if (!this.state.isChecked) {
+            return;
+        }
+        this.props.dispatch(addTodo(this.state.input.val));
+        this.setState({isChecked: false,val:" "})
+    };
+
+    toggleChange = (event) => {
+      this.setState({
+        isChecked: !this.state.isChecked},
+      {input: {val: event.target.value}}
+    ) // flip boolean value
+
+
+    };
+
+
+    render = () => {
+        const submitTodoButton =  <Button bsStyle="primary" type="submit"><Glyphicon glyph="pencil" /></Button>;
+
+        return (
+            <div>
+                <form onSubmit={this.todoSubmit.bind(this)}>
+                <label><input  type="checkbox"   ref="input" bsSize="large"  checked={this.state.isChecked}  onChange={this.toggleChange} />   Check Here..!
+                   {submitTodoButton}   </label>
+                </form>
+            </div>
+        )
+    }
+}
+
+AddTodo = connect()(AddTodo);
```

Comment:
Might want to read about mapStateToProps to see how to get the values out of Redux's state for these components